### ISSUE TITLE
Misleading argument name on seq2seq module

### ISSUE THREAD

**Issue by Conchylicultor** [tensorflow/tensorflow#4205]
Not really major issues, but can make lose some time.

For the functions [sequence_loss_by_example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/seq2seq.py#L980) and [sequence_loss](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/seq2seq.py#L1026), the name of the parameter `softmax_loss_function` seems to imply that only softmax based loss are compatibles which is not true (for instance `tf.nn.sigmoid_cross_entropy_with_logits` works too). Maybe the name was that just to indicate the default behavior but I find it misleading.

It shouldn't be really difficult to change (as there are very few references in the repository). I can send a pull request if needed.
